File: src (to delete)/main/java/com/hodvidar/insuratradeflow/mvc/persistance/dao/InsurancePolicyDao.java
package com.hodvidar.insuratradeflow.mvc.persistance.dao;

import com.hodvidar.insuratradeflow.mvc.business.domain.PolicyStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "insurance_policies")
@NoArgsConstructor
@AllArgsConstructor // for tests
@Getter
@Setter
public class InsurancePolicyDao {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private PolicyStatus status; // Enum for ACTIVE, INACTIVE

    @Column(nullable = false)
    private LocalDate startDate;

    @Column(nullable = false)
    private LocalDate endDate;

    @Column(nullable = false, updatable = false)
    @Temporal(TemporalType.TIMESTAMP)
    private LocalDateTime creationDateTime;

    @Column(nullable = false)
    @Temporal(TemporalType.TIMESTAMP)
    private LocalDateTime updateDateTime;

    @PrePersist
    protected void onCreate() {
        LocalDateTime now = LocalDateTime.now();
        creationDateTime = now;
        updateDateTime = now;
    }

    @PreUpdate
    protected void onUpdate() {
        updateDateTime = LocalDateTime.now();
    }
}

File: backend/src/node/persistence/daos/insurance-policy.dao.ts
import {Column, CreateDateColumn, Entity, PrimaryGeneratedColumn, UpdateDateColumn} from 'typeorm';

@Entity()
export class InsurancePolicyDao {
    @PrimaryGeneratedColumn()
    id?: number;

    @Column()
    name: string;

    @Column({type: 'varchar'})
    status: string;  // stored as string in DB

    @Column({type: 'date', name: 'start_date'})
    startDate: Date;

    @Column({type: 'date', name: 'end_date'})
    endDate: Date;

    @CreateDateColumn({type: 'timestamp', name: 'creation_date_time'})
    creationDateTime?: Date;

    @UpdateDateColumn({type: 'timestamp', name: 'update_date_time'})
    updateDateTime?: Date;
}